pkg/sdk/event_tables_validations_gen.go

```diff
@@ -0,0 +1,197 @@
+package sdk
+
+import "errors"
+
+var (
+	_ validatable = new(CreateEventTableOptions)
+	_ validatable = new(ShowEventTableOptions)
+	_ validatable = new(DescribeEventTableOptions)
+	_ validatable = new(AlterEventTableOptions)
+)
+
+func (opts *CreateEventTableOptions) validate() error {
+	if opts == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if !ValidObjectIdentifier(opts.name) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	if everyValueSet(opts.OrReplace, opts.IfNotExists) && *opts.OrReplace && *opts.IfNotExists {
+		errs = append(errs, errOneOf("OrReplace", "IfNotExists"))
+	}
+	if valueSet(opts.CopyGrants) && !valueSet(opts.OrReplace) {
+		errs = append(errs, errors.New("CopyGrants requires OrReplace"))
+	}
+	return JoinErrors(errs...)
+}
+
+func (opts *ShowEventTableOptions) validate() error {
+	if opts == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	return JoinErrors(errs...)
+}
+
+func (v *EventTableClusteringAction) validate() error {
+	if v == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if ok := anyValueSet(
+		v.ClusterBy,
+		v.ResumeRecluster,
+		v.SuspendRecluster,
+		v.DropClusteringKey,
+	); !ok {
+		errs = append(errs, errAtLeastOneOf("ClusterBy", "ResumeRecluster", "SuspendRecluster", "DropClusteringKey"))
+	}
+	if ok := exactlyOneValueSet(
+		v.ClusterBy,
+		v.ResumeRecluster,
+		v.SuspendRecluster,
+		v.DropClusteringKey,
+	); !ok {
+		errs = append(errs, errExactlyOneOf("ClusterBy", "ResumeRecluster", "SuspendRecluster", "DropClusteringKey"))
+	}
+	return errors.Join(errs...)
+}
+
+func (v *EventTableSearchOptimizationAction) validate() error {
+	if v == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if ok := anyValueSet(
+		v.Add,
+		v.Drop,
+	); !ok {
+		errs = append(errs, errAtLeastOneOf("Add", "Drop"))
+	}
+	if ok := exactlyOneValueSet(
+		v.Add,
+		v.Drop,
+	); !ok {
+		errs = append(errs, errExactlyOneOf("Add", "Drop"))
+	}
+	return errors.Join(errs...)
+}
+
+func (v *EventTableSet) validate() error {
+	if v == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if ok := anyValueSet(
+		v.DataRetentionTimeInDays,
+		v.MaxDataExtensionTimeInDays,
+		v.ChangeTracking,
+		v.Comment,
+	); !ok {
+		errs = append(errs, errAtLeastOneOf("DataRetentionTimeInDays", "MaxDataExtensionTimeInDays", "ChangeTracking", "Comment"))
+	}
+	if ok := exactlyOneValueSet(
+		v.DataRetentionTimeInDays,
+		v.MaxDataExtensionTimeInDays,
+		v.ChangeTracking,
+		v.Comment,
+	); !ok {
+		errs = append(errs, errExactlyOneOf("DataRetentionTimeInDays", "MaxDataExtensionTimeInDays", "ChangeTracking", "Comment"))
+	}
+	return errors.Join(errs...)
+}
+
+func (v *EventTableUnset) validate() error {
+	if v == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if ok := anyValueSet(
```

Comment:
It seems like you have written many of these validations by hand. They should be all added in the def file (for most of the cases the def file is enough to generate all the validations and this is this case).

Furthermore, our generator is `smart` so in addition to the generated validation it also generates the test headers for all the validations, so you can remember which ones you have to implement.

Reply:
regenerated validations